mempool/src/pool.rs

```diff
@@ -790,6 +801,25 @@ where
             .collect()
     }
 
+    fn collect_txs(
+        &self,
+        mut tx_accumulator: Box<dyn TransactionAccumulator>,
+    ) -> Vec<SignedTransaction> {
+        if self.store.is_empty() {
+            return Vec::new();
+        }
+        let mut tx_iter = self.store.txs_by_descendant_score.values().flatten();
+        while !tx_accumulator.done() {
+            if let Some(next_tx_id) = tx_iter.next() {
+                let next_tx = self.store.txs_by_id.get(next_tx_id).expect("tx to exist");
```

Comment:
This seems like a good place to implement an iterator for the store so that you don't have to expect like this... not a priority though. It'll become something like a multi-index container.

Reply:
Added a `TODO` for this.